### DISCUSSION TITLE
How to set an environmental variable `OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT`?

### DISCUSSION THREAD

**Discussion by indoyoon** [open-telemetry/opentelemetry-rust#2214] [Q&A] [answered]
I want to set an environmental variable `OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT`, which is defined at [OpenTelemetry documentation](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#span-limits).

For example, in Python SDK, we can set the variable to limit the length of the attribute. However I could not find the reference for doing the same thing in our Rust SDK.

Is this not possible at the moment in Rust SDK? Thank you.

**Comment by cijothomas** [comment, marked as answer]
https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#span-limits is *not* OpenTelemetry documentation, but it is OTel specification. Different language implementations are at different levels in terms on implementation of each spec aspect, especially the support for ENV variables as OTel spec lists ENV variable as *optional*.

OTel Rust does not support this env variable today. So you have to manually set it when building TracerProvider.

**Comment by indoyoon** [comment]
Thank you. In that case, we will implement the feature and create a pull request.
